activerecord/lib/active_record/destroy_association_async_job.rb

```diff
@@ -23,9 +23,12 @@ def perform(
         raise DestroyAssociationAsyncError, "owner record not destroyed"
       end
 
-      association_model.where(association_primary_key_column => association_ids).find_each do |r|
```

Comment:
Most likely we also need to tweak `owner = owner_class.find_by(owner_class.primary_key.to_sym => owner_id)` otherwise `blog_posts` is being queried without `blog_id` (out implied tenant_key) 

But we can postpone it until we have a way to reveal actual problem using models with a db-level composite primary key

Reply:
Let's postpone it until we have the Car/Book model in, then we can add test coverage and get that in as well.